test/e2e/service.go

```diff
@@ -1099,6 +1099,89 @@ var _ = framework.KubeDescribe("Services", func() {
 			expectedSuccess := nodes.Items[n].Name == readyHostName
 			jig.TestHTTPHealthCheckNodePort(publicIP, healthCheckNodePort, "/healthz", expectedSuccess)
 		}
+
+		By("changing ESIPP annotation off and checking behaviour")
+		// Toggle ESIPP annotation off and check client ip and healthchecks
+		svc = jig.UpdateServiceOrFail(svc.Namespace, svc.Name, func(svc *api.Service) {
+			svc.ObjectMeta.Annotations[service.AnnotationExternalTraffic] =
+				service.AnnotationValueExternalTrafficGlobal
+		})
+		success := false
+		for start := time.Now(); time.Since(start) < kubeProxyLagTimeout; time.Sleep(15 * time.Second) {
+			By("reading clientIP using the TCP service's NodePort")
+			content := jig.GetHTTPContent(nodeIP, tcpNodePort, kubeProxyLagTimeout, "/clientip")
+			clientIP := content.String()
+			framework.Logf("ClientIP detected by target pod using NodePort is %s", clientIP)
+			By("reading clientIP using the TCP service's service port via its external VIP")
+			content = jig.GetHTTPContent(ingressIP, svcTcpPort, kubeProxyLagTimeout, "/clientip")
+			clientIP = content.String()
+			framework.Logf("ClientIP detected by target pod using VIP:SvcPort is %s", clientIP)
```

Comment:
can you only log on failure? like ip Failf("returned from service different from vip"), or something? All this is noise in test logs if the failure is elsewhere in this test. Ideally we'd just see something like:
```
Step: changing ESIPP annotation off and checking cross node proxy
Step: changing ESIPP back on and checking source ip
Step: something else
```

and when we see a failure: 
```
Step: one
Step: two
  Failed to verify source ip, VIP returned blah service returned Blah
Step: three
```

I'll take a follow up if you don't want to do it in this one, but this test is already gigantic.